linebot/api.py

```diff
@@ -1148,6 +1149,244 @@ def get_bot_info(self, timeout=None):
 
         return BotInfo.new_from_json_dict(response.json)
 
+    def create_audience_group(self, audience_group_name, audiences=[],
+                              is_ifa=False, timeout=None):
+        """Create an audience group.
+
+        https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group
+
+        :param str audience_group_name: The audience's name
+        :param list audiences: An array of user IDs or IFAs
+        :param bool is_ifa: true | false
+        :return: audience group id
+        """
+        response = self._post(
+            '/v2/bot/audienceGroup/upload',
+            data=json.dumps({
+                "description": audience_group_name,
+                "isIfaAudience": is_ifa,
+                "audiences": [{"id": audience} for audience in audiences[:10000]],
+            }),
+            timeout=timeout
+        )
+        audience_group_id = response.json.get('audienceGroupId')
+        if 10000 < len(audiences):
+            self.add_audiences_to_audience_group(audience_group_id, audiences[10000:])
```

Comment:
`add_audiences_to_audience_group` is let me confuse, please help me to discuss it.

Reply:
It is considered that Line API has the limitation (1000 per request) on  audience IDS uploading. uploading. 
https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group

Reply:
I think you don't need to judge it because error response will return `AUDIENCE_GROUP_COUNT_MAX_OVER` in `detail` property.

refer to: https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group-error-response

Reply:
modified